tests/src/python/test_qgstaskmanager.py

```diff
@@ -261,6 +262,36 @@ def finished_multiple_value_result(e, results):
         self.assertEqual(result_value, 5)
         self.assertEqual(result_statement, 'whoo')
 
+    def testTaskFromFunctionWithSubTaskCompletedIsCalledOnce(self):  # spellok
+        """ test that when a parent task has subtasks it does emit taskCompleted only once"""
+
+        self.finished = 0
+        self.completed = 0
+
+        def _on_finished():
+            self.finished += 1
+
+        def _on_completed():
+            self.completed += 1
+
+        task = QgsTask.fromFunction('test task', run_no_result, on_finished=_on_finished)
+        task.taskCompleted.connect(_on_completed)
+        spy = QSignalSpy(task.taskCompleted)
+        sub_task_1 = QgsTask.fromFunction('test subtask 1', run_no_result, on_finished=_on_finished)
+        sub_task_2 = QgsTask.fromFunction('test subtask 2', run_no_result, on_finished=_on_finished)
+        task.addSubTask(sub_task_1, [], QgsTask.ParentDependsOnSubTask)
+        task.addSubTask(sub_task_2, [], QgsTask.ParentDependsOnSubTask)
+
+        QgsApplication.taskManager().addTask(task)
+        while task.status() not in [QgsTask.Complete, QgsTask.Terminated]:
+            pass
```

Comment:
I can't get this test to pass locally unless I insert QCoreApplication.processEvents() here instead of pass - can you confirm?

Reply:
Yes and no: the test hangs with pass but fails with processEvents:

```
Traceback (most recent call last):
  File "/home/xxx/dev/QGIS/tests/src/python/test_qgstaskmanager.py", line 292, in testTaskFromFunctionWithSubTaskCompletedIsCalledOnce
    self.assertEqual(self.finished, 3)
AssertionError: 0 != 3
```